packages/s-deploy/src/deploy.ts

```diff
@@ -20,21 +22,52 @@ class Deploy {
   }
 
   async run() {
-    const { resource, deployFile, deployArgs, envName, debugMode, workspace } = this.props;
+    const { cmd, args } = await this.generateDeployCommand();  
+    const { workspace } = this.props;
+    this.logger.info(`Execute command: ${cmd} ${args.join(' ')}\n\n`)
+    try {
+      const subprocess = execa(cmd, args, {
+        shell: true,
+        stripFinalNewline: false,
+        cwd: workspace,
+      });
+      subprocess.stdout?.pipe(process.stdout);
+      subprocess.stderr?.pipe(process.stderr);
+      await subprocess;
+    } catch(e) {
+      throw new Error(_.get(e, 'originalMessage'));
+    }
+  }
+
+  async generateDeployCommand() {
+    const { resource, deployFile, deployArgs, envName, infraStackName, debugMode, workspace } = this.props;
    
     const cmd = resource ? `s ${resource} deploy` : 's deploy';
     let args = ['--use-local', '--assume-yes', '--skip-push'];
-
-    if (envName) {
-      args.push(`--env ${envName}`);
+    
+    const isDevsV3 = (fileName: string) => {
+      const syaml = yaml.load(fs.readFileSync(fileName, 'utf8'));
+      return _.get(syaml, 'edition') === '3.0.0';
     }
-  
+
+    let manifest
     if (deployFile) {
-      const manifest = path.resolve(workspace, deployFile);
+      manifest = path.resolve(workspace, deployFile);
       if (!fs.existsSync(manifest)) {
         throw new Error(`The deploy manifest [${manifest}] does not exist`)
       }
       args.push(`-t ${deployFile}`)
+    } else {
+      manifest = path.resolve(workspace, 's.yaml');
```

Comment:
args.push

Reply:
No need to modify the args here because deployFile is not specified. The purpose is to let ServerlessDevs use the default s.yaml.